### ISSUE TITLE
MatrixFree partition_partition gives segmentation fault in the unit test

### ISSUE THREAD

**Issue by davydden** [dealii/dealii#3134]
```
$ git diff
diff --git a/tests/matrix_free/parallel_multigrid_adaptive.cc b/tests/matrix_free/parallel_multigrid_adaptive.cc
index 0de815a..5758aad 100644
--- a/tests/matrix_free/parallel_multigrid_adaptive.cc
+++ b/tests/matrix_free/parallel_multigrid_adaptive.cc
@@ -62,8 +62,8 @@ public:
   {
     const QGauss<1> quad (n_q_points_1d);
     typename MatrixFree<dim,number>::AdditionalData addit_data;
-    addit_data.tasks_parallel_scheme = MatrixFree<dim,number>::AdditionalData::none;
-    addit_data.tasks_block_size = 3;
+    addit_data.tasks_parallel_scheme = MatrixFree<dim,number>::AdditionalData::partition_partition;
+    //addit_data.tasks_block_size = 3;
     addit_data.level_mg_handler = level;
     addit_data.mpi_communicator = MPI_COMM_WORLD;
     ConstraintMatrix constraints;
```
leads to segmentation fault during `data.reinit()` for `mpirun=3` both in `debug` and `release`. I don't think it is expected behaviour.

@kronbichler any ideas?

**Comment by davydden** [comment]
`partition_color` works fine here.

**Comment by kronbichler** [comment]
@davydden Can you please try the fix in #3135?
